rust/crd/src/lib.rs

```diff
@@ -48,6 +53,41 @@ pub struct SparkApplicationStatus {
     pub phase: String,
 }
 
+#[derive(Clone, Debug, Default, Deserialize, Merge, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SparkStorageConfig {
+    #[serde(default)]
+    pub data: PvcConfig,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SparkConfig {
+    pub resources: Option<Resources<SparkStorageConfig, NoRuntimeLimits>>,
+}
+
+impl SparkConfig {
+    fn default_resources() -> Resources<SparkStorageConfig, NoRuntimeLimits> {
```

Comment:
I suggest making the default resources depending on the role (job, driver, executor)
The job needs basically nothing (maybe 50m/100m cou, 1gb ram), default for driver i would suggest something like 100m/1 CPU and 2gb ram and executors 1/4 cpu and 4gb ram
Just throwing in some numbers to give an idea, no need to stick to them exactly

Reply:
Good idea! Yes, I will do that

Reply:
Done.